client/app/account/user/agreements/details/user-agreements-details.controller.js

```diff
@@ -1,54 +1,68 @@
-angular.module("UserAccount").controller("UserAccountAgreementsDetailsController", [
-    "$stateParams",
-    "$q",
-    "$log",
-    "UserAccount.services.agreements",
-    "Alerter",
-    "$translate",
-    "User",
-    function ($stateParams, $q, $log, Service, Alerter, $translate, User) {
-        "use strict";
+angular
+    .module("UserAccount")
+    .controller("UserAccountAgreementsDetailsController", class UserAccountAgreementsDetailsController {
+        constructor ($q, $stateParams, $translate, Alerter, CGV_AGREEMENT_ID, OvhApiMe) {
+            this.$q = $q;
+            this.$stateParams = $stateParams;
+            this.$translate = $translate;
+            this.Alerter = Alerter;
+            this.CGV_AGREEMENT_ID = CGV_AGREEMENT_ID;
+            this.OvhApiMe = OvhApiMe;
+        }
 
-        const CGV_AGREEMENT_ID = 1635;
-
-        this.$ngInit = () => {
+        $onInit () {
             this.accepted = false;
             this.loading = true;
             this.confirmed = false;
             this.alreadyAccepted = false;
+            this.api = {
+                me: {
+                    agreements: this.OvhApiMe.Agreements().v6()
+                }
+            };
 
-            $q
-                .all([Service.getAgreement($stateParams.id), Service.getContract($stateParams.id), User.getUser()])
-                .then((result) => {
-                    this.agreement = result[0];
-                    this.contract = result[1];
-                    this.isIndividual = result[2].legalform === "individual";
-                    this.alreadyAccepted = this.agreement.agreed === "ok";
+            return this.$q
+                .all({
+                    contractAgreement: this.fetchContractAgreement(this.$stateParams.id),
+                    contract: this.fetchContract(this.$stateParams.id),
+                    user: this.fetchUser()
```

Comment:
For the three fetch methods, shoudn't they be declared before use ?

Reply:
Not needed. We are in a ES6 `class`.